server.js

```diff
@@ -4,20 +4,91 @@ const cors = require("cors");
 const app = express();
 
 app.use(cors());
+//these two middleware functions allow an 
+//Express app to parse the request body of an HTTP request
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }))
 
-const welcomeMessage = {
-  id: 0,
-  from: "Bart",
-  text: "Welcome to CYF chat system!",
-};
+// const welcomeMessage = {
+//   id: 0,
+//   from: "Bart",
+//   text: "Welcome to CYF chat system!",
+// };
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+const messages = [];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+// Enables a new message to be created 
+app.post("/messages", function (request, response) {
+  const { from, text } = request.body;
+  if (!from || !text) {
+    return response.status(400).send("Please provide a name and a message");
+  }
+
+  const newMessage = {
+    id: messages.length,
```

Comment:
this method of selecting an id can lead to duplicate id values.

Imaging you had a set of 20 messages, where the last message is id:20. You delete message 19. When you next send a message, it will also be given id:20